lib/my_inject.rb

```diff
@@ -0,0 +1,60 @@
+class MyArray < Array
+
+  def my_inject(initial = nil, sym=nil, &block)
+    dup = self.dup
+    if initial == nil
+      initial = dup.shift
+      #When there is no block and only one argument,
+      #this argument is assigned to initial.
+      #but I need it to be symbol.
+      #The next 4 lines is the only way I could find to deal with it.
+      #Any advise is highly appreciated
+    elsif !block && (initial.is_a? Symbol)
+      sym = initial
+      initial = dup.shift
+    end
+
+    if block
+      dup.each {|el| initial = block.call(initial, el)}
+      initial
+    elsif sym
+      dup.each {|el| initial = initial.send(sym, el)}
+      initial
+    else
+      fail "ERROR!"
+    end
+  end
+
+  def my_inject_rec(initial=nil,sym=nil,index=0,&block)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for my_inject_rec is too high. [21.75/15]